package/src/core/with-plugins.ts

```diff
@@ -14,12 +14,7 @@ type WithPluginsParams<TPlugins extends NonEmptyArray<AnyPlugin>> = {
 	hooks: ExtendedHooks<TPlugins>;
 };
 
-type WithPluginsReturn<Extensions> = Omit<
-	Extensions,
-	"name" | "plugins" | "hooks"
-> & {
-	hooks: AstroIntegration["hooks"];
-};
+export type WithPluginsReturn<Extensions> = Extensions & Omit<AstroIntegration, 'name'>;
```

Comment:
Is this really doing the same as the previous one?

Reply:
It is.

The omit for the extension is now done inline on the return type just below cause the declaration emits the call to this type, not the result of it.

The omit for `AstroIntegration` is used here in case new fields get added to core.